src/Validator/UniqueUrlValidator.php

```diff
@@ -36,14 +38,21 @@ public function __construct(PageManagerInterface $manager)
 
     public function validate($value, Constraint $constraint): void
     {
-        if (!$value instanceof PageInterface) {
-            $this->context->addViolation('The page is not valid, expected a PageInterface');
-
+        if (!$constraint instanceof UniqueUrl) {
+            throw new UnexpectedTypeException($constraint, UniqueUrl::class);
```

Comment:
I think this is not correct, you cant receive another Constraint, otherwise this code wont be called.

This exception class is used when the value is not the type ypu expect to be (ex: I expect the value to be a Page and I am getting string).

Reply:
All the other symfony validators do it

https://github.com/symfony/symfony/blob/870eeb975feb1abb4b8a1722e1fd57beeab2b230/src/Symfony/Component/Validator/Constraints/LanguageValidator.php#L32-L34

Reply:
Yeah, I'm doing the same in my project. To me it seems correct @jordisala1991